collections.js

```diff
@@ -118,8 +118,36 @@ LinkedList.prototype.find = function(value) {
     node = node.next;
   }
   return null;
+};
+
+//------------------------------------------------------------------------------
+// Simple Queue.
+
+function SimpleQueue() {
+	this.q_ = [];
 }
 
+SimpleQueue.prototype = {
+  enqueue: function(item) {
+    this.q_.push(item);
+    return this;
+  },
+
+  dequeue: function() {
+    return this.q_.shift();
+  },
+
+  empty: function() {
+    return this.q_.length === 0;
+  },
+  
+  flush: function() {
```

Comment:
clear() for consistency with other collections here and JS builtin collections (Map, Set)